docs(functions): add tokens doc (#913)
scaleway/docs-content: compute/functions/how-to/create-auth-token-from-console.mdx

---
meta:
  title: How to create and manage an an authentication token from the console
  description: This page explains how to create and manage an authentication token from the console
content:
  h1: How to create and manage an authentication token from the console
  paragraph: This page explains how to create and manage an authentication token from the console
dates:
  validation: 2022-10-14
  posted: 2022-10-14
---

This page shows you how to create and manage an authentication token from the console.

<Message type="requirement">

  - You have an account and are logged into the [Scaleway console](https://console.scaleway.com)
  - You have [created a functions namespace](/compute/functions/how-to/create-a-function/)
  - You have [created a function](/compute/functions/how-to/create-a-function/)

</Message>

## Creating athentication tokens from the console 

> 1. Click **Functions** in the **Serverless** section of the side menu. The functions page displays.

Reviewer: ```suggestion
1. Click **Functions** in the **Serverless** section of the side menu. The Serverless Functions page displays.
```

> 2. Click the relevant functions namespace to see a list of functions in this namespage.

Reviewer: ```suggestion
2. Click the relevant functions namespace to see a list of functions in this namespace.
```

> 3. Click the functions name in the **Functions** tab. The functions details' display. 

Reviewer: ```suggestion
3. Click the name of the relevant function in the **Functions** tab. The function's details display. 
```

4. Click the **Functions Settings** tab. A list of your functions settings displays. 

> 5. Click **Generate token** in the **Tokens** sections of the settings page. A pop-up displays.

Reviewer: ```suggestion
5. Click **Generate token** in the **Tokens** panel. A pop-up displays.
```

    <Lightbox src="scaleway-functions-generate-token.webp" alt="" />

> 6. Enter the token description and set an expiration date for the token. Optionally tick the box **No expiration date**, to have a token with unlimited validity. Then click **Generate token**

Reviewer: ```suggestion
6. Enter the token description and set an expiration date for the token. Optionally, tick the box **No expiration date**, to have a token with unlimited validity. Then click **Generate token**
```

> 6. Enter the token description and set an expiration date for the token. Optionally, tick the box **No expiration date**, to have a token with unlimited validity. Then click **Generate token**

Reviewer 2: ```suggestion
6. Enter the token description and set an expiration date for the token. Optionally, tick the box **No expiration date**, to have a token with unlimited validity. Then click **Generate token**.
```

7. Copy the token from the pop-up and store it in a safe environment. 

    <Message type="important">

    The token key will only be available for copy once. Make sure to store it securely before leaving the page.

    </Message>

## Using tokens

> To use your token, you have to add the generated token to a `x-auth-token`-header

Reviewer 3: ```suggestion
To use your token, you have to add the generated token to a `x-auth-token`-header.
```

A **private** function observes this behaviour:

* If a call is made without the` x-auth-token`-header, the call is rejected (status code 403)
* If the `x-auth-token`-header is provided, the token is validated using a public key attached to the namespace.

For example, to execute a private function by providing a JWT using `curl`, you may run the following command:
```
curl -H "X-AUTH-TOKEN: <generated-token>" <your-function-host>
```

## Deleting tokens 

Tokens will expire by default on their given expiry date. You can delete them manually if they are no longer needed. 

> 1. Click **Functions** in the **Serverless** section of the side menu. The functions page displays.

Reviewer: ```suggestion
1. Click **Functions** in the **Serverless** section of the side menu. The Serverless Functions page displays.
```

> 2. Click the relevant functions namespace to see a list of functions in this namespage.

Reviewer: ```suggestion
2. Click the relevant functions namespace to see a list of functions in this namespace.
```

> 3. Click the functions name in the **Functions** tab. The functions details' display. 

Reviewer: ```suggestion
3. Click the name of the relevant function in the **Functions** tab. The function's details display. 
```

4. Click the **Functions Settings** tab. A list of your functions settings displays. 

> 5. Click <Icon name="delete" /> in the **Tokens** sections of the settings page. A pop-up displays.

Reviewer: ```suggestion
5. Click <Icon name="delete" /> in the **Tokens** panel. A pop-up displays.
```
